src/main/java/org/fenixedu/academic/util/AcademicPhoneNumberHandler.java

```diff
@@ -0,0 +1,119 @@
+package org.fenixedu.academic.util;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.fenixedu.academic.domain.contacts.EmailAddress;
+import org.fenixedu.academic.domain.contacts.MobilePhone;
+import org.fenixedu.academic.domain.contacts.PartyContact;
+import org.fenixedu.academic.domain.contacts.Phone;
+import org.fenixedu.academic.domain.contacts.PhysicalAddress;
+import org.fenixedu.academic.domain.contacts.WebAddress;
+import org.fenixedu.academic.util.PhoneUtil.PhoneNumberHandler;
+
+import com.google.i18n.phonenumbers.NumberParseException;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.PhoneNumberUtil.PhoneNumberFormat;
+import com.google.i18n.phonenumbers.PhoneNumberUtil.PhoneNumberType;
+import com.google.i18n.phonenumbers.Phonenumber.PhoneNumber;
+
+public class AcademicPhoneNumberHandler implements PhoneNumberHandler {
+
+    protected static final PhoneNumberUtil PHONE_UTIL = PhoneNumberUtil.getInstance();
+
+    private static final Map<Class<? extends PartyContact>, ContactResolver> DEFAULT_RESOLVERS = new HashMap<>();
+
+    static {
+        DEFAULT_RESOLVERS.put(EmailAddress.class, (pc) -> ((EmailAddress) pc).getValue());
+        DEFAULT_RESOLVERS.put(MobilePhone.class, (pc) -> ((MobilePhone) pc).getNumber());
+        DEFAULT_RESOLVERS.put(Phone.class, (pc) -> ((Phone) pc).getNumber());
+        DEFAULT_RESOLVERS.put(PhysicalAddress.class, (pc) -> ((PhysicalAddress) pc).getAddress());
+        DEFAULT_RESOLVERS.put(WebAddress.class, (pc) -> ((WebAddress) pc).getUrl());
+    }
+
+    private static final Collection<PhoneNumberType> FIXED_NUMBERS;
```

Comment:
You can init the constant in the same line.
`private static final Collection<PhoneNumberType> FIXED_NUMBERS = new ArrayList<>();`